sdk/core/azure-core/src/http/url.cpp

```diff
@@ -271,3 +206,10 @@ std::string Url::GetAbsoluteUrl() const
   full_url += GetRelativeUrl();
   return full_url;
 }
+
+std::unordered_set<unsigned char> Url::defaultNonUrlEncodeChars
+    = {'a', 'b', 'c',  'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q',
+       'r', 's', 't',  'u', 'v', 'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H',
+       'I', 'J', 'K',  'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y',
+       'Z', '0', '1',  '2', '3', '4', '5', '6', '7', '8', '9', '-', '.', '_', '~', '"', '!',
```

Comment:
why `"` appear twice?

shouldn't we encode `"`?

Reply:
The default should be to encode any character other than A-Z a-z 0-9 and these 4 symbols: - . _ ~
These are the only characters allowed to be unencoded in any part of a URL.
In theory, encoding any other characters is not bad to do; it may be encoding more than what has to be encoded.
If the caller wants to opt-out of the "conservative encoding", then the caller can pass the additional characters to NOT encode.
This is the thinking behand this API design.

Reply:
then we will have to append query like this `url.AppendQuery(key, Encode(value, "/:@?"));`.

we have 140 append query operations just in [blob service protocol layer](https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp), it looks not good to repeat string `/:@?` 140 times.